ethabi/src/decoder.rs

```diff
@@ -202,9 +206,34 @@ fn decode_param(param: &ParamType, slices: &[[u8; 32]], offset: usize) -> Result
 				new_offset,
 			};
 
-			Ok(result)
-		}
-	}
+            Ok(result)
+        }
+        ParamType::Tuple(ref t) => {
+            let tuple_len_offset = if param.is_dynamic() {
+                let tuple_offset_slice = try!(peek(slices, offset));
+                (as_u32(tuple_offset_slice)? / 32) as usize
+            } else {
+                offset
+            };
+
+            let len = t.len();
+            let mut tokens = Vec::with_capacity(len);
+            let tuple_slices = &slices[tuple_len_offset..];
+            let mut new_offset = 0;
+            for i in 0..len {
+                let res = decode_param(&t[i], &tuple_slices, new_offset)?;
+                new_offset = new_offset + 1;
+                tokens.push(res.token);
+            }
+
+            let result = DecodeResult {
+                token: Token::Tuple(tokens),
+                new_offset: new_offset + 1,
```

Comment:
Is there a reason for this extra `+ 1`? I don't see that in the other param decoder branches.